src/reorder.h

```diff
@@ -0,0 +1,23 @@
+#ifndef REORDER_H
+#define REORDER_H
+
+#include <vector>
+#include <cassert>
+
+/*
+  Reorder a vector using a sorting index
+*/
+template <typename value_t, typename index_t>
+void reorder(std::vector<value_t> &arr, const std::vector<index_t> &order) {
+
+  assert(arr.size()==order.size());
+  
+  std::vector<value_t> arr_sorted(order.size());
+#pragma omp parallel for schedule(static, 10*1024)
```

Comment:
Where does the size for the OMP chunk come from?

Reply:
I don't really have a very good reason for that, apart from not wanting it to use a lot of threads or incur scheduling overhead for tiny loops.